lib/datadog/core/telemetry/logging.rb

```diff
@@ -65,6 +74,20 @@ def error(description)
 
           log!(event)
         end
+
+        private
+
+        # A constant string representing the exception
+        def message_for_telemetry(exception)
+          if exception.respond_to?(:telemetry_message)
+            exception.telemetry_message
+          # For the 150+ `Errno` error classes, we set the telemetry message as an instance variable.
+          # We don't want to create/modify such a large number of classes for a small feature.
+          # Note: Errno classes are subclasses of SystemCallError.
+          elsif exception.is_a?(SystemCallError) && exception.instance_variable_defined?(:@telemetry_message)
+            exception.instance_variable_get(:@telemetry_message)
+          end
+        end
```

Comment:
Minor: I wonder if we should just get rid of the attr_acessor and simplify the code to always use the instance_variable_defined + instance_variable_get...